Sources/MosaiqueAssetsPicker/Views/AssetDetailCell.swift

```diff
@@ -128,17 +128,20 @@ final class AssetDetailCell: UICollectionViewCell, AssetDetailCellBindable {
 
     func setDownloading(_ isDownloading: Bool) {
         if isDownloading {
-            let _spinner: UIActivityIndicatorView
-            if let c = self.spinner {
-                _spinner = c
-            } else {
-                _spinner = UIActivityIndicatorView(style: .whiteLarge)
-                self.spinner = _spinner
+
+            let spinner = self.spinner ?? {
+                let spinner = UIActivityIndicatorView(style: .whiteLarge)
+                self.spinner = spinner
+                return spinner
+                }()
+
+            if spinner.superview != contentView {
+                contentView.addSubview(spinner)
             }
```

Comment:
I thought, we should check the spinner has been added to the content view every time. because isDownloading flag turns false, the spinner will remove from superview.

Reply:
Ah I didn't notice we do not set `spinner` to `nil` in case of `prepareForReuse`. It's definitely more robust this way !